engines/stark/stark.cpp

```diff
@@ -247,6 +247,11 @@ void StarkEngine::updateDisplayScene() {
 	// Clear the screen
 	_gfx->clearScreen();
 
+	// Quit to main menu screen, if needed
+	if (_userInterface->hasQuitToMainMenuRequest()) {
```

Comment:
This check feels like it does not belong here. Couldn't it be in `StarkEngine::mainLoop` with the quit check and the location change check?